File: appverse-backend/src/main/java/com/appverse/service/AppService.java
package com.appverse.service;

import com.appverse.dto.AppDTO;

import java.util.List;

public interface AppService {
    AppDTO createApp(AppDTO appDTO, Long developerId);
    AppDTO updateApp(Long id, AppDTO appDTO, Long developerId);
    AppDTO getAppById(Long id);
    List<AppDTO> getAllApps();
    List<AppDTO> getAppsByCategory(Long categoryId);
    List<AppDTO> getAppsByDeveloper(Long developerId);
    List<AppDTO> searchApps(String keyword);
    List<AppDTO> getTrendingApps();
    List<AppDTO> getTopRatedApps();
    void deleteApp(Long id, Long developerId);
    void deactivateApp(Long id);
}

File: appverse-frontend/src/services/appService.js
import api from './api';

const appService = {
  getAllApps: async () => {
    const response = await api.get('/apps');
    return response.data;
  },

  getAppById: async (id) => {
    const response = await api.get(`/apps/${id}`);
    return response.data;
  },

  getAppsByCategory: async (categoryId) => {
    const response = await api.get(`/apps/category/${categoryId}`);
    return response.data;
  },

  searchApps: async (keyword) => {
    const response = await api.get(`/apps/search?keyword=${encodeURIComponent(keyword)}`);
    return response.data;
  },

  getTrendingApps: async () => {
    const response = await api.get('/apps/trending');
    return response.data;
  },

  getTopRatedApps: async () => {
    const response = await api.get('/apps/top-rated');
    return response.data;
  },

  downloadApp: async (appId) => {
    const response = await api.post(`/apps/${appId}/download`);
    return response.data;
  },

  getCategories: async () => {
    const response = await api.get('/categories');
    return response.data;
  },
};

export default appService;
